pkg/local_object_storage/pilorama/types.go

```diff
@@ -0,0 +1,52 @@
+package pilorama
+
+import (
+	"errors"
+	"math"
+)
+
+// Timestamp is an alias for integer timestamp type.
+// TODO: remove after the debugging.
+type Timestamp = uint64
+
+// Node is used to represent nodes.
+// TODO: remove after the debugging.
+type Node = uint64
+
+// Meta represents arbitrary meta information.
+// TODO: remove after the debugging or create a proper interface.
+type Meta struct {
+	Time  Timestamp
+	Items []KeyValue
+}
+
+// KeyValue represents a key-value pair.
+type KeyValue struct {
+	Key   string
+	Value []byte
+}
+
+// Move represents a single move operation.
+type Move struct {
+	Parent Node
+	Meta
+	// Child represents the ID of a node being moved. If zero, new ID is generated.
+	Child Node
+}
+
+// LogMove represents log record for a single move operation.
+type LogMove struct {
+	Move
+	HasOld bool
+	Old    nodeInfo
+}
+
+const (
+	// RootID represents the ID of a root node.
+	RootID = 0
+	// TrashID is a parent for all removed nodes.
+	TrashID = math.MaxUint64
+)
+
+// ErrNotFound is returned when node id is not found or there is no nodes corresponding to the filepath.
+var ErrNotFound = errors.New("object not found")
```

Comment:
It would be nice to have a way to know such error on client side without `strings.Contains`

Reply:
It is more of a gRPC issue, the error is transmitted by network, I doubt it is possible to have something more convenient.

Also, this error is actually for missing _tree_, changed the error message.